clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordsTest.java

```diff
@@ -31,32 +31,129 @@
+        Iterator<ConsumerRecord<Integer, String>> iterator = records.iterator();
 
-        Map<TopicPartition, List<ConsumerRecord<Integer, String>>> records = new LinkedHashMap<>();
+        int recordCount = 0;
+        int partitionCount = 0;
+        int currentPartition = -1;
+
+        while (iterator.hasNext()) {
+            ConsumerRecord<Integer, String> record = iterator.next();
+
+            if (record.partition() == emptyPartitionIndex) {
+                fail("Partition " + emptyPartitionIndex + " is not empty");
+            }
+
+            // Check if we have moved to a new partition
+            if (currentPartition != record.partition()) {
+                // Increment the partition count as we have encountered a new partition
+                partitionCount++;
+                // Update the current partition to the new partition
+                currentPartition = record.partition();
+            }
 
-        String topic = "topic";
-        records.put(new TopicPartition(topic, 0), new ArrayList<>());
-        ConsumerRecord<Integer, String> record1 = new ConsumerRecord<>(topic, 1, 0, 0L, TimestampType.CREATE_TIME,
-            0, 0, 1, "value1", new RecordHeaders(), Optional.empty());
-        ConsumerRecord<Integer, String> record2 = new ConsumerRecord<>(topic, 1, 1, 0L, TimestampType.CREATE_TIME,
-            0, 0, 2, "value2", new RecordHeaders(), Optional.empty());
-        records.put(new TopicPartition(topic, 1), Arrays.asList(record1, record2));
-        records.put(new TopicPartition(topic, 2), new ArrayList<>());
-
-        ConsumerRecords<Integer, String> consumerRecords = new ConsumerRecords<>(records);
-        Iterator<ConsumerRecord<Integer, String>> iter = consumerRecords.iterator();
-
-        int c = 0;
-        for (; iter.hasNext(); c++) {
-            ConsumerRecord<Integer, String> record = iter.next();
-            assertEquals(1, record.partition());
             assertEquals(topic, record.topic());
-            assertEquals(c, record.offset());
+            assertEquals(currentPartition, record.partition());
+            assertEquals(recordCount % recordSize, record.offset());
+            assertEquals(recordCount % recordSize, record.key());
+            assertEquals(String.valueOf(recordCount % recordSize), record.value());
+
+            recordCount++;
         }
-        assertEquals(2, c);
+
+        // Including empty partition
+        assertEquals(partitionSize, partitionCount + 1);
+    }
+
+    @Test
+    public void testRecordsWithNullTopic() {
+        String nullTopic = null;
+        ConsumerRecords<Integer, String> consumerRecords = ConsumerRecords.empty();
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> consumerRecords.records(nullTopic));
+        assertEquals("Topic must be non-null.", exception.getMessage());
+    }
+
+
+    @Test
+    public void testRecords() {
```

Comment:
Could you add test for `records(TopicPartition)`?

Reply:
I will do it 😸